algoliasearch-core/src/main/java/com/algolia/search/util/AlgoliaUtils.java

```diff
@@ -11,6 +16,10 @@
 
 public class AlgoliaUtils {
 
+
+  // ZoneRules is immutable and threadsafe, but getRules method consumes a lot of memory during load testing.
+  private static final ZoneRules ZONE_RULES_UTC = ZoneOffset.UTC.getRules();
```

Comment:
Would you mind moving this static declaration closer to `nowUTC()`.

Also, if you'd like, you could make this comment part of the `nowUTC()` comment to avoid splitting the information in two places.